gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/ReadOption.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.datastore;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * Specifies options for read operations in Datastore, namely getting/fetching entities and running
+ * queries.
+ */
+public abstract class ReadOption implements Serializable {
+
+  private static final long serialVersionUID = -4406964829189800528L;
+
+  /**
+   * Specifies eventual consistency for reads from Datastore.
+   */
+  public static final class EventualConsistency extends ReadOption {
+
+    private static final long serialVersionUID = -6959530217724666172L;
+
+    private final boolean eventualConsistency;
+
+    private EventualConsistency(boolean eventualConsistency) {
+      this.eventualConsistency = eventualConsistency;
+    }
+
+    public boolean isEventual() {
+      return eventualConsistency;
+    }
+  }
+
+  ReadOption() {}
+
+  /**
+   * Returns a {@code ReadOption} that specifies eventual consistency.
+   */
+  public static EventualConsistency eventualConsistency() {
+    return new EventualConsistency(true);
+  }
+
+  static Map<Class<? extends ReadOption>, ReadOption> asImmutableMap(ReadOption... options) {
```

Comment:
Did you consider using an enum instead of `Class` as we do in storage and other services? I see that this is a little different as the option map does not reach the spi layer but gets transformed to dedicated lookup/read options. Anyway it might be worth considering enums to have all possible options in one place (no need to look for classes that extend `ReadOption`.